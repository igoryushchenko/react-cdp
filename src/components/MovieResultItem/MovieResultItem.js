// @flow
import React from 'react';
import { useHistory } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { showMovieDetailsAction } from '../../store/actions';
import type { MovieProps } from '../../types/MovieProps';

const MovieResultItem = ({
  id, title, release_date, poster_path, genres,
}: MovieProps) => {
  const dispatch = useDispatch();
  const history = useHistory();

  const handleMovieSelect = (movieId) => {
    dispatch(showMovieDetailsAction(movieId));
    history.push(`/film/${movieId}`);
  };

  return (
    <div id={id} className="card col-lg-4">
        <img className="card-img-top"
          alt={`Picture for movie: ${title}`}
          src={poster_path}
             onClick={() => handleMovieSelect(id)}
        />
      <div className="card-body">
        <div className="card-title">
          <span className="h4">{title}</span>
          <span className="border border-dark rounded float-right p-1">{release_date.slice(0, 4)}</span>
        </div>
        <div>{genres.join(' & ')}</div>
      </div>
    </div>
  );
};

export default MovieResultItem;
