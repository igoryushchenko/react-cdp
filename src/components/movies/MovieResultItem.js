import React from 'react';
import PropTypes from 'prop-types';

const MovieResultItem = ({ id, title, release_date, poster_path, genres, handleMovieSelect }) => {
  return (
    <div id={id} className="card col-lg-4">
        <img className="card-img-top"
          alt={`Picture for movie: ${title}`}
          src={poster_path}
             onClick={() => handleMovieSelect(id)}
        />
      <div className="card-body">
        <div className="card-title">
          <span className="h4">{title}</span>
          <span className="border border-dark rounded float-right p-1">{release_date.slice(0, 4)}</span>
        </div>
        <div>{genres.join(' & ')}</div>
      </div>
    </div>
  );
};

MovieResultItem.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  poster_path: PropTypes.string,
  release_date: PropTypes.string,
  genres: PropTypes.arrayOf(PropTypes.string),
  handleMovieSelect: PropTypes.func
};

export default MovieResultItem;