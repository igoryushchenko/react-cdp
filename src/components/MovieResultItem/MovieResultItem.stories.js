import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { storiesOf } from '@storybook/react';
import MovieResultItem from './MovieResultItem';
import storyWrapper from '../../shared/storyWrapper';

const movie = {
  id: 412302,
  title: 'Gemini',
  tagline: '',
  vote_average: 10,
  vote_count: 3,
  release_date: '2018-03-30',
  poster_path: 'https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg',
  overview: 'A heinous crime tests the complex relationship between a tenacious personal assistant and her Hollywood starlet boss. As the assistant travels across Los Angeles to unravel the mystery, she must stay one step ahead of a determined policeman and confront her own understanding of friendship, truth and celebrity.',
  budget: 0,
  revenue: 0,
  genres: [
    'Mystery',
    'Thriller',
  ],
  runtime: 92,
};

storiesOf('MovieResultItem', module)
  .addDecorator(storyWrapper)
  .add('with movie details', () => <MovieResultItem key={movie.id} {...movie}/>);
