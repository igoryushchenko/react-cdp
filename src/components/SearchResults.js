import React from 'react';
import PropTypes from 'prop-types';
import MovieResultItem from './MovieResultItem';
import MovieResultEmpty from './MovieResultEmpty';

const SearchResults = ({movieData}) => {
  return (
    <>
      <div className="row">
        {movieData.total > 0 ? (movieData.data.map(
            mov => <MovieResultItem key={mov.id} {...mov} />))
        : (<MovieResultEmpty />)
        }
      </div>
    </>
  );
};

SearchResults.propTypes = {
  movieData: PropTypes.shape({
    total: PropTypes.number,
    data: PropTypes.array
  })
};

export default SearchResults;
