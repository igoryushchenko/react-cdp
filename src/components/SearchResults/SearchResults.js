import React from 'react';
import PropTypes from 'prop-types';
import MovieResultItem from '../MovieResultItem/MovieResultItem';
import MovieResultEmpty from '../MovieResultEmpty/MovieResultEmpty';

const SearchResults = ({ movieData, handleMovieSelect }) => (
    <>
      <div className="row">
        {movieData.total > 0 ? (movieData.data.map(
          (mov) => <MovieResultItem key={mov.id} {...mov} handleMovieSelect={handleMovieSelect}/>,
        ))
          : (<MovieResultEmpty />)
        }
      </div>
    </>
);

SearchResults.propTypes = {
  movieData: PropTypes.shape({
    total: PropTypes.number,
    data: PropTypes.array,
  }),
  handleMovieSelect: PropTypes.func,
};

export default SearchResults;
