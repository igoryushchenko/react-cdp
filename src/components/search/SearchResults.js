import React from 'react';
import PropTypes from 'prop-types';
import MovieResultItem from '../movies/MovieResultItem';
import MovieResultEmpty from '../movies/MovieResultEmpty';

const SearchResults = ({movieData, handleMovieSelect}) => {
  return (
    <>
      <div className="row">
        {movieData.total > 0 ? (movieData.data.map(
            mov => <MovieResultItem key={mov.id} {...mov} handleMovieSelect={handleMovieSelect}/>))
        : (<MovieResultEmpty />)
        }
      </div>
    </>
  );
};

SearchResults.propTypes = {
  movieData: PropTypes.shape({
    total: PropTypes.number,
    data: PropTypes.array
  }),
  handleMovieSelect: PropTypes.func
};

export default SearchResults;
