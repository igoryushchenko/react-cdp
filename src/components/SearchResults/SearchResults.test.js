import React from 'react';
import { shallow, mount, render } from 'enzyme';
import * as Redux from 'react-redux';
import SearchResults from './SearchResults';
import MovieResultItem from '../MovieResultItem/MovieResultItem';

describe('SearchResults component', () => {
  let searchResults;

  beforeAll(() => {
    searchResults = {
      total: 2,
      data: [
        {
          id: 412302,
          title: 'Gemini',
          tagline: '',
          vote_average: 10,
          vote_count: 3,
          release_date: '2018-03-30',
          poster_path: 'https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg',
          overview: 'A heinous crime tests the complex relationship between a tenacious personal assistant and her Hollywood starlet boss. As the assistant travels across Los Angeles to unravel the mystery, she must stay one step ahead of a determined policeman and confront her own understanding of friendship, truth and celebrity.',
          budget: 0,
          revenue: 0,
          genres: [
            'Mystery',
            'Thriller',
          ],
          runtime: 92,
        },
        {
          id: 141052,
          title: 'Justice League',
          tagline: '',
          vote_average: 6.4,
          vote_count: 3960,
          release_date: '2017-11-15',
          poster_path: 'https://image.tmdb.org/t/p/w500/eifGNCSDuxJeS1loAXil5bIGgvC.jpg',
          overview: "Fuelled by his restored faith in humanity and inspired by Superman's selfless act, Bruce Wayne and Diana Prince assemble a team of metahumans consisting of Barry Allen, Arthur Curry and Victor Stone to face the catastrophic threat of Steppenwolf and the Parademons who are on the hunt for three Mother Boxes on Earth.",
          budget: 500000000,
          revenue: 655894816,
          genres: [
            'Action',
            'Adventure',
            'Fantasy',
            'Science Fiction',
          ],
          runtime: 120,
        },
      ],
    };
  });

  it('should render text "No films found"', () => {
    const searchResultsEmpty = {
      data: [],
    };
    const component = shallow(<SearchResults movieData={searchResultsEmpty} />);
    expect(component).toMatchSnapshot();
    expect(render(<SearchResults movieData={searchResultsEmpty} />).text()).toEqual('No films found');
  });

  it('should be render two elements', () => {
    const component = shallow(<SearchResults movieData={searchResults} />);
    expect(component).toMatchSnapshot();
  });

  it('should be render two component - MovieResultItem', () => {
    const useDispatchSpy = jest.spyOn(Redux, 'useDispatch');
    useDispatchSpy.mockImplementation(() => (cb) => (cb));
    const component = mount(<SearchResults movieData={searchResults} />);
    expect(component.find(MovieResultItem).length).toBe(2);
  });
});
