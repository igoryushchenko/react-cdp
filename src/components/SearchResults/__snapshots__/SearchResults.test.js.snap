// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component should be render two elements 1`] = `ShallowWrapper {}`;

exports[`SearchResults component should render text "No films found" 1`] = `ShallowWrapper {}`;
